cloud/services/publicloadbalancers/publicloadbalancers.go

```diff
@@ -133,6 +134,23 @@ func (s *Service) Reconcile(ctx context.Context, spec interface{}) error {
 						},
 					},
 				},
+				OutboundRules: &[]network.OutboundRule{
+					{
+						Name: to.StringPtr("OutboundRuleUDPTCP"),
```

Comment:
if anyone can think of a better name let me know

Reply:
How about "EnableOutboundSNAT" as the name?

Reply:
is it okay to not mention UDP/TCP in the name?

Reply:
Strictly speaking UDP and TCP aren't the only protocol types, I imagine the `LoadBalancerOutboundRuleProtocolAll` const will allow for more than that; so perhaps mentioning UDP/TCP exclusively is actually misleading.

Reply:
according to https://docs.microsoft.com/en-us/azure/load-balancer/load-balancer-outbound-rules-overview#proto they are: "Use the following parameter to set the protocol to TCP and UDP:"

Reply:
🤕 🤕

Reply:
I think we should have some comments here as to why this is needed - the comments on the commit are basically what the code should have such that when someone looks at this code in the future they don't go "why?"

Now, the name of the rule helps a bit but some more details will help the next poor soul who looks at this and asks "why?"

Reply:
added a comment